test/e2e/util.go

```diff
@@ -54,6 +54,21 @@ func waitForPodRunning(c *client.Client, id string) {
 	}
 }
 
+func waitForPodNotPending(c *client.Client, id string) {
```

Comment:
Consider adding a timeout argument, otherwise this functionality ends up being pushed into the code that uses this function.